pkg/operator/ceph/object/spec.go

```diff
@@ -912,3 +922,93 @@ func renderProbe(cfg rgwProbeConfig) (string, error) {
 
 	return writer.String(), nil
 }
+
+func (c *clusterConfig) addDNSNamesToRGWServer() (string, error) {
+	if (c.store.Spec.Hosting == nil) || len(c.store.Spec.Hosting.DNSNames) <= 0 {
+		return "", nil
+	}
+	if !c.clusterInfo.CephVersion.IsAtLeastReef() {
+		return "", errors.New("rgw dns names are supported from ceph v18 onwards")
+	}
+
+	// add default RGW service name to dns names
+	dnsNames := c.store.Spec.Hosting.DNSNames
+	dnsNames = append(dnsNames, domainNameOfService(c.store))
+
+	// add custom endpoints from zone spec if exists
+	if c.store.Spec.Zone.Name != "" {
+		zone, err := c.context.RookClientset.CephV1().CephObjectZones(c.store.Namespace).Get(c.clusterInfo.Context, c.store.Spec.Zone.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		dnsNames = append(dnsNames, zone.Spec.CustomEndpoints...)
+	}
+
+	// validate dns names
+	var hostNames []string
+	for _, dnsName := range dnsNames {
+		hostName, err := GetHostnameFromEndpoint(dnsName)
+		if err != nil {
+			return "", err
+		}
+		hostNames = append(hostNames, hostName)
+	}
+
+	// remove duplicate host names
+	checkDuplicate := make(map[string]bool)
+	removeDuplicateHostNames := []string{}
+	for _, hostName := range hostNames {
+		if _, ok := checkDuplicate[hostName]; !ok {
+			checkDuplicate[hostName] = true
+			removeDuplicateHostNames = append(removeDuplicateHostNames, hostName)
+		}
+	}
+
+	return cephconfig.NewFlag("rgw dns name", strings.Join(removeDuplicateHostNames, ",")), nil
+}
+
+func GetHostnameFromEndpoint(endpoint string) (string, error) {
+	if len(endpoint) == 0 {
+		return "", fmt.Errorf("endpoint cannot be empty string")
+	}
+
+	// if endpoint doesn't end in '/', Ceph adds it
+	// Rook can do this also to get more accurate error results from this function
+	if !strings.HasSuffix(endpoint, "/") {
+		endpoint = endpoint + "/"
+	}
+
+	// url.Parse() requires a protocol to parse the host name correctly
+	if !strings.Contains(endpoint, "://") {
+		endpoint = "http://" + endpoint
+	}
+
+	// "net/url".Parse() assumes that a URL is a "path" with optional things surrounding it.
+	// For Ceph RGWs, we assume an endpoint is a "hostname" with optional things surrounding it.
+	// Because of this difference in fundamental assumption, Rook needs to adjust some endpoints
+	// used as input to url.Parse() to allow the function to extract a hostname reliably. Also,
+	// Rook needs to look at several parts of the url.Parse() output to identify more failure scenarios
+	parsedURL, err := url.Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
+
+	logger.Debugf("parsedURL : %#v", parsedURL)
```

Comment:
I think perhaps this debug line isn't very useful during rook runtime. Maybe a single debug line after the cli flag is rendered would be enough, and would be less log lines.